### PR TITLE
WIP: Draft Zapier docs

### PR DESCRIPTION
**Pull request by esethna**
Need to add:
- troubleshooting guide
- instructions on how to use use the app as invite only (need to test once we flip the switch)

Reminder that the formatting will look different on Sphinx (especially for code blocks and nested lists).

### CODE REVIEW

**Comment by esethna** [comment]
Ready for review. Added troubleshooting for account connection.

File: source/integrations/zapier.rst [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  f. **Callback URLs**: ``https://zapier.com/dashboard/auth/oauth/return/App27274API/``
+3. Click **Save** to create the application. You will be provided with a **Client ID** and **Client Secret**. Save these values to connect Zapier in the steps below.
+
+.. image:: ../images/zapier-oauth.png
+
+
+Create a Zap
+~~~~~~~~~~~~~~~~~~~~~~~~~
+1. Click and accept this invitation to our Zapier app: https://zapier.com/developer/invite/27274/233f5629a5c215f542747a9c6d395b98/
+2. On your `Zapier dashboard <https://zapier.com/app/dashboard>`_ click **Make a Zap!**.
```
Comment (lindy65) [inline source/integrations/zapier.rst]: At this point, I realised I needed a Zapier account.  Perhaps we need to clarify that?

File: source/integrations/zapier.rst [hunk 2]
```diff
@@ -0,0 +1,147 @@
+Zapier and Mattermost
+===================
+
```
Comment (lfbrock) [inline source/integrations/zapier.rst]: Maybe add a sentence clarifying what Zapier is

File: source/integrations/zapier.rst [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+Integrate Mattermost with over 500 supported apps on Zapier, including `GitHub <https://zapier.com/zapbook/github/>`_, `Jira <https://zapier.com/zapbook/jira/>`_, `Wufoo <https://zapier.com/zapbook/wufoo/>`_, `Salesforce <https://zapier.com/zapbook/salesforce/>`_, `Twitter <https://zapier.com/zapbook/twitter/>`_, `Gmail <https://zapier.com/zapbook/gmail/>`_ and `many more <https://zapier.com/zapbook/>`_. Zapier integration triggers new Mattermost messages for events in other apps. This documentation includes:
```
Comment (lfbrock) [inline source/integrations/zapier.rst]: Can we clarify what Zapier is a little more, for those who don't know? 

Maybe add in "Zapier is a service that ..... " before the listed integrations. 

Also it should probably be "The Zapier integration triggers"



"Zapier integration triggers new Mattermost messages for events in other apps."

File: source/integrations/zapier.rst [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+- `Zapier Setup Guide <https://docs.mattermost.com/integrations/zapier.html#zapier-setup-guide>`_ - **Instructions on registering the Zapier app on your server and creating a zap**
+- `Message Formatting Tips <https://docs.mattermost.com/integrations/zapier.html#message-formatting-tips>`_ - **Tips from the Mattermost team on formatting Zapier integration messages**
+- `Troubleshooting Guide <https://docs.mattermost.com/integrations/zapier.html#troubleshooting-guide>`_ - **Advice on troubleshooting common setup issues**
+
+
+Zapier Setup Guide
+--------------------------------
+Zapier is authorized using OAuth2.0. The setup guide requires that a System Admin register the Zapier app on their Mattermost server and can then optionally allow any users on the system to create Zapier integrations.
```
Comment (lfbrock) [inline source/integrations/zapier.rst]: I think Lindy is right, we should mention you need to sign up for a Zapier account at the beginning here (it's really only any user with a zapier account who can set it up, not any user)

File: source/integrations/zapier.rst [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+.. image:: ../images/zapier-oauth.png
+
+
+Create a Zap
+~~~~~~~~~~~~~~~~~~~~~~~~~
+1. Click and accept this invitation to our Zapier app: https://zapier.com/developer/invite/27274/233f5629a5c215f542747a9c6d395b98/
+2. On your `Zapier dashboard <https://zapier.com/app/dashboard>`_ click **Make a Zap!**.
+3. **Trigger App**: Events in this app will trigger new messages in Mattermost.
+  a. **Select a Trigger App** or choose from one of Zapier’s built-in apps that will trigger new messages in Mattermost. If the app you’re looking to connect isn’t supported on Zapier, consider firing in-app events to a Gmail account and then connecting Gmail to Mattermost using Zapier.
```
Comment (lfbrock) [inline source/integrations/zapier.rst]: "or choose" does not make sense? (selecting a trigger app typically IS choosing from one of Zapier's built-in apps?)

File: source/integrations/zapier.rst [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  c. **Connect the Action Account**. Click **Connect a New Account** and enter the following fields:
+    1. **Mattermost URL**: This is the URL you use to access your Mattermost site. Do not include a slash at the end of the URL and do not append a team to the end of the server URL. For example, ``https://pre-release.mattermost.com/core`` is the entire URL to the Contributors team on our pre-release server. The **Mattermost URL** entered here would be ``https://pre-release.mattermost.com``.
+    2. **Client ID/Secret**: If Zapier has been enabled as an OAuth application as per the steps above, then these values can also be found by navigating to one of your Mattermost teams, then **Main Menu** > **Integrations** > **OAuth 2.0 Applications**. Click **Show Secret** next to the Zapier app, then obtain the Client ID and Client Secret. 
+    3. **Login to Mattermost**. After completing the above fields you will be prompted to login to your Mattermost account if you are not logged in already. If you’re having trouble connecting then please read our `troubleshooting guide <https://docs.mattermost.com/integrations/zapier.html#troubleshooting-guide>`_.
```
Comment (lfbrock) [inline source/integrations/zapier.rst]: Missing the step to click "Allow" on the "Zapier would like to connect to your Mattermost user account" page

File: source/integrations/zapier.rst [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+5. **Mattermost Action**: Connect your Mattermost Account and then specify posting details.
+  a. **Select the Action App**. Search for “Mattermost”.
+  b. **Select the Action Event**. Select **Post a Message**. The Mattermost team plans to expand the actions available here.
+  c. **Connect the Action Account**. Click **Connect a New Account** and enter the following fields:
+    1. **Mattermost URL**: This is the URL you use to access your Mattermost site. Do not include a slash at the end of the URL and do not append a team to the end of the server URL. For example, ``https://pre-release.mattermost.com/core`` is the entire URL to the Contributors team on our pre-release server. The **Mattermost URL** entered here would be ``https://pre-release.mattermost.com``.
+    2. **Client ID/Secret**: If Zapier has been enabled as an OAuth application as per the steps above, then these values can also be found by navigating to one of your Mattermost teams, then **Main Menu** > **Integrations** > **OAuth 2.0 Applications**. Click **Show Secret** next to the Zapier app, then obtain the Client ID and Client Secret.
```
Comment (lfbrock) [inline source/integrations/zapier.rst]: why "can also be found"

File: source/integrations/zapier.rst [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+    2. **Client ID/Secret**: If Zapier has been enabled as an OAuth application as per the steps above, then these values can also be found by navigating to one of your Mattermost teams, then **Main Menu** > **Integrations** > **OAuth 2.0 Applications**. Click **Show Secret** next to the Zapier app, then obtain the Client ID and Client Secret. 
+    3. **Login to Mattermost**. After completing the above fields you will be prompted to login to your Mattermost account if you are not logged in already. If you’re having trouble connecting then please read our `troubleshooting guide <https://docs.mattermost.com/integrations/zapier.html#troubleshooting-guide>`_. 
+  d. **Message Post Details**: Specify the formatting of the messages and the team/channel where messages will post.
+    1. **Team**: Choose the team where new messages will post. The dropdown should contain all teams you have access to on Mattermost.
```
Comment (lfbrock) [inline source/integrations/zapier.rst]: Are System Admins able to see all teams (even those they don't belong to?)

File: source/integrations/zapier.rst [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+    2. **Channel**: Choose the channel where new messages will post. Zapier cannot post into Direct Message channels, but all other Channels and Private Groups should appear in the dropdown.
```
Comment (lfbrock) [inline source/integrations/zapier.rst]: is this "all Public Channels, and the Private Groups you belong to"? Private Groups you're not a part of don't show up right?

File: source/integrations/zapier.rst [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+Enable Zapier on your Mattermost Instance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The first time you set up Zapier on your Mattermost instance you will be required to enable an OAuth 2.0 application which can be accessible by everyone on your server. Your System Admin must execute these steps.
+
+Enable OAuth 2.0
+```````````````````````````
+1. Go to the **Main Menu** > **System Console**
+2. Under **Integrations** > **Custom Integrations**
+  a. Set `Enable OAuth 2.0 Service Provider <https://docs.mattermost.com/administration/config-settings.html#enable-oauth-2-0-service-provider>`_ to **True**.
+  b. If you’d like to allow Zapier integrations to post with customizable usernames and profile pictures, then set `Enable webhooks and slash commands to override usernames <https://docs.mattermost.com/administration/config-settings.html#enable-webhooks-and-slash-commands-to-override-usernames>`_ and `Enable webhooks and slash commands to override profile picture icons <https://docs.mattermost.com/administration/config-settings.html#enable-webhooks-and-slash-commands-to-override-profile-picture-iconss>`_ to **True**.
```
Comment (esethna) [inline source/integrations/zapier.rst]: True, updated config settings with a commit on this branch

**Comment by esethna** [comment]
Thanks @lindy65 and @lfbrock, tweaks made if you want to take another look before merging?

**Pull request merged**